src/pages/home/userPosts.tsx

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from 'react';
+import UserPostCard from '../../components/UserPostCard';
+import { getPostImage } from '../../util/getImages';
+import { getPostsByUser } from '../../util/getPosts';
+
+import './userPosts.css';
+
+interface IProps {
+  userId: number;
+}
+
+const UserPosts = ({ userId }: IProps) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [postsInformation, setPostsInformation] = useState([]);
+  const [imageInformation, setImageInformation] = useState({
+    download_url: '',
+  });
+  useEffect(() => {
+    getPostImage(userId).then((imageData) => {
+      setImageInformation(imageData);
+    });

```

Comment:
This probably makes more sense to include in the User card so you can fetch the image for each post rather than a single image for all of them

Reply:
Yes you are right, here i was trying to match postId with images in a different api endpoint because the jsonplaceholder did not provide a valid image, but i did not apply a logic to match different images.